.github/workflows/bump-and-release.yml

```diff
@@ -0,0 +1,115 @@
+        required: true
+        default: true
+        type: boolean
+      connectors:
+        description: 'Choose the Connector(s)'
+        required: true
+        type: choice
+        default: 'All'
+        options:
+          - All
+          - Utilities
+          - Comscore
+          - Conviva
+          - SidelaodedSubtitle
+jobs:
+  Bump-And-Release:
+    runs-on: macos-latest
+    outputs:
+      previousVersion: ${{ steps.store_previous_version.outputs.previousVersion }}
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+      - id: store_previous_version
+        run: |
+          previousVersion=$(ruby -r "./THEOplayer-Connector-Version.rb" -e "print_theoplayer_connector_version")
+          echo "previousVersion=$previousVersion" >> "$GITHUB_OUTPUT"
+      - name: Update the version.json file
+        if: inputs.version != ''
+        run: sh update_version_json.sh -v ${{ inputs.version }}
+      - name: Commit and push the changes made to the version.json file
+        if: inputs.version != ''
+        run: |
+          git add version.json
+          git commit -m "update version.json"
+          git push
+      - name: Add and push new git tag
+        if: inputs.version != ''
+        run: |
+          git tag ${{ inputs.version }}
+          git push origin ${{ inputs.version }}
+      - name: ${{ (inputs.dryRun == true && 'Validate') || 'Release' }} on Cocoapods
+        env:
+          COCOAPODS_TRUNK_TOKEN: ${{ secrets.COCOAPODS_TRUNK_TOKEN }}
+        run: |
+          # make the command
+          command="${{ (inputs.dryRun == true && 'spec lint') || 'trunk push' }}"
+          # make the list of connectors
+          prefix="THEOplayer-Connector"
+          connectors=()
+          if [ "${{ inputs.connectors }}" == "All" ]; then
+            files=(*)
+            for file in ${files[*]}
+            do
+              if [[ $file == *.podspec ]]; then
+                # we add the Utilities later at index 0
+                if [[ $file == *Utilities.podspec ]]; then
+                  continue
+                fi
+                # blacklisted podspecs
+                if [[ $file == *VerizonMedia.podspec || $file == *Nielsen.podspec ]]; then
+                  continue
+                fi
+                # Conviva validation depends on Utilities being available, which does not get published in a dry run
+                # => skip Conviva if dry run
+                if [[ $file == *Conviva.podspec && ${{ inputs.dryRun == true }} ]]; then
+                  continue
+                fi
+                connectors+=("$file")
+              fi
+            done
+            connectors=("$prefix-Utilities.podspec" "${connectors[@]}")
+          else
+            connectors+=("$prefix-${{ inputs.connectors }}.podspec")
+          fi
+          for connector in ${connectors[*]}
+          do
+            pod $command $connector --verbose --allow-warnings
+          done
+  Cleanup:
```

Comment:
I am missing the point (couldn't find it in the code) in which scenario this is running.
How/When is this job triggered?
Asking, because I see it is removing some tags at some point, which could break then the published releases, no?

Reply:
Line 92, the if condition.
`if: (inputs.dryRun == true && inputs.version != '') && (success() || failure())`
If the version tag is passed, we don't do cleanup since there will be no version bump/tag.
If it's not a dry run (so release) then we don't do cleanup obviously.
So we only do the cleanup if dry running and if version is passed.
Finally, the success() || failure() makes sure that the cleanup still happens if an error occurs in the previous job.